tests/src/Kernel/CollaboraMediaAccessTest.php

```diff
@@ -0,0 +1,410 @@
+use Drupal\media\MediaInterface;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+use Drupal\user\Entity\Role;
+use Drupal\user\RoleInterface;
+
+/**
+ * Tests access for media operations and routes.
+ */
+class CollaboraMediaAccessTest extends KernelTestBase {
+
+    use MediaTypeCreationTrait;
+    use UserCreationTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected static $modules = [
+        'collabora_online',
+        'user',
+        'media',
+        'field',
+        'image',
+        'system',
+        'file',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp(): void {
+        parent::setUp();
+
+        $this->installEntitySchema('user');
+        $this->installEntitySchema('file');
+        $this->installSchema('file', 'file_usage');
+        $this->installEntitySchema('media');
+        $this->installConfig([
+            'field',
+            'system',
+            'user',
+            'image',
+            'file',
+            'media'
+        ]);
+
+        $this->createMediaType('file', ['id' => 'document']);
+
+        // Consume the user id 1.
+        $this->createUser();
+    }
+
+    /**
+     * Verifies basic assumptions for this test.
+     */
+    public function testBasicAssumptions(): void {
+        // Normally, when media module is installed, media_install() will grant
+        // the 'view media' permission to anonymous and authenticated roles.
+        // In a kernel test this does not happen.
+        $this->assertSame([], Role::load(RoleInterface::ANONYMOUS_ID)->getPermissions());
+        $this->assertSame([], Role::load(RoleInterface::AUTHENTICATED_ID)->getPermissions());
+
+        // The first user created in a test method is not user 1.
+        $user = $this->createUser();
+        $this->assertSame(2, (int) $user->id());
+        $this->assertFalse($user->hasPermission('view media'));
+    }
+
+    /**
+     * Tests access for users with admin-like permissions.
+     */
+    public function testAdminAccess(): void {
+        /** @var \Drupal\Core\Session\AccountInterface[] $accounts */
+        $accounts = [
+            'admin' => $this->createUser(admin: TRUE),
+            'media admin' => $this->createUser(['administer media']),
+        ];
+
+        /** @var \Drupal\media\MediaInterface[] $media_entities */
+        $media_entities = [
+            'published document' => $this->createMediaEntity('document'),
+            'unpublished document' => $this->createMediaEntity('document', [
+                'status' => 0,
+            ]),
+        ];
+
+        // The admin accounts should have access to all operations.
+        $this->assertEntityAccess(
+            [],
+            $accounts,
+            $media_entities,
+            [
+                'view',
+                'preview in collabora',
+                'edit in collabora',
+            ],
+            negate: TRUE,
```

Comment:
This is a shortcut to avoid to write all the permission results, and I don't like it at all.
We are already using an atypic yet useful approach to testing permissions/access, but let's try to keep it sane.
Remove the negate and explicitly write all the operations and all the routes.